### PR TITLE
LevelZero support for correctable & uncorrectable RAS counters

### PR DESCRIPTION
**Pull request by sidjana**
- Relates to #3363 

The past support for Level-Zero RAS counters lacked awareness of RAS error types (correctable v/s uncorrectable). Only correctable counters were supported. 

This PR introduces the awareness of both types of RAS errors and extends the list of LevelZero IOGroup signals to include both types.

### CODE REVIEW

**Review by cmcantalupo — DISMISSED** [review]

File: libgeopmd/src/LevelZero.cpp [hunk 1]
```diff
@@ -159,12 +159,12 @@ namespace geopm
         // TODO: When additional device types such as FPGA, MCA, and Integrated GPU are supported by GEOPM
         // This should be changed to a more general loop iterating over type and caching appropriately
         for (unsigned int gpu_idx = 0; gpu_idx < m_num_gpu; gpu_idx++) {
-            ras_domain_cache(gpu_idx);
             frequency_domain_cache(gpu_idx);
             power_domain_cache(gpu_idx);
             perf_domain_cache(gpu_idx);
             engine_domain_cache(gpu_idx);
             temperature_domain_cache(gpu_idx);
+            ras_domain_cache(gpu_idx);
```
Comment (dannosliwcd) [inline libgeopmd/src/LevelZero.cpp:167]: If order matters, can you add a comment about why `ras_domain_cache` needs to be last?

File: libgeopmd/src/LevelZero.cpp [hunk 2]
```diff
@@ -481,53 +481,80 @@ namespace geopm
...
+		// Cache the RAS errorset handle into index - [subdevice Id][error type]
+                else if (property.type == ZES_RAS_ERROR_TYPE_CORRECTABLE ||
+                         property.type == ZES_RAS_ERROR_TYPE_UNCORRECTABLE) {
+                   int ras_idx = property.subdeviceId * M_NUM_ERROR_TYPE +
+                                 (property.type == ZES_RAS_ERROR_TYPE_CORRECTABLE ?
+                                  M_ERROR_TYPE_CORRECTABLE : M_ERROR_TYPE_UNCORRECTABLE);
+                   m_devices.at(device_idx).subdevice.ras_domain.at(ras_idx) = handle;
+                   ++ras_handle_count;
+#ifdef GEOPM_DEBUG
+		   std::cerr << "Level-0 RAS: Inserted handle for subdevice# "<<property.subdeviceId<<", error type# "<<property.type<<", device# "<<device_idx<<"\n";
```
Comment (dannosliwcd) [inline libgeopmd/src/LevelZero.cpp]:
```suggestion
		   std::cerr << "Note: <geopm> Level-0 RAS: Inserted handle for subdevice# "<<property.subdeviceId<<", error type# "<<property.type<<", device# "<<device_idx<<"\n";
```
Should make it clear this debug note comes from GEOPM. Or if it was only there for initial development debug, maybe delete it.

File: libgeopmd/src/LevelZero.cpp [hunk 3]
```diff
@@ -605,71 +632,153 @@ namespace geopm
...
                         GEOPM_ERROR_RUNTIME,
                         "LevelZero::" + std::string(__func__) +
                         ": Sysman failed to get RAS counters",
                         __LINE__);
-
-        std::array<uint64_t, ZES_MAX_RAS_ERROR_CATEGORY_COUNT> temp;
-        for (unsigned int err_idx = 0; err_idx < ZES_MAX_RAS_ERROR_CATEGORY_COUNT; err_idx++)
-            temp[err_idx] = pState.category[err_idx];
-        return temp;
+	return pState.category.at(errorcat);
```
Comment (dannosliwcd) [inline libgeopmd/src/LevelZero.cpp]: This looks like we fully populate the `pState.category` array then return just a single element for the requested category. Does that cancel any benefits we get from combining RAS queries in GEOPM batch reads?

File: libgeopmd/src/LevelZero.cpp (hunk 1 above)
Comment (sidjana) [reply libgeopmd/src/LevelZero.cpp:167]: The only reason why ras_domain_cache is placed at the end is due to the undocumented ongoing coding practice of adding the "most recently added L0 signal" towards the end of the caching order. No technical reason.

File: libgeopmd/src/LevelZero.cpp (hunk 2 above)
Comment (sidjana) [reply libgeopmd/src/LevelZero.cpp]: Good catch. This was debugging code intended only for initial development. Deleting line.

File: libgeopmd/src/LevelZero.cpp (hunk 3 above)
Comment (sidjana) [reply libgeopmd/src/LevelZero.cpp]: Note that the full array - `pState.category` is populated by L0 sysman, not PlatformIO. Each time a single element needs to be accessed, the entire array needs to be referenced. This overhead of populating all of contents of the `category` array is embedded within the execution model of L0. In fact, this latest commit removes the additional 'for' loop by removing the extra overhead of caching into 'temp' (on top of the aforementioned L0 internal overhead).

**Review by dannosliwcd — APPROVED** [review]
Please open a tech debt issue referring back here about the `pState.category` batch_read discussion

**Review by dannosliwcd — APPROVED** [review]

**Pull request merged**